File: AmazonClone/Domain/Entities/Product.cs
using System.Reflection.Metadata;

namespace AmazonClone.Domain.Entities
{
    public class Product : Entity
    {
        public string name { get; set; }
        public float price { get; set; }
        public string description { get; set; }
        public ICollection<ProductPhoto> photos { get; set; }
        public ICollection<Comment> comments { get; set; }
    }
}

File: client/src/app/interfaces/product.ts
export interface Product {
  id: string;
  name: string;
  price: number;
  description: string;
  comments: Comment[];
  productCategories: ProductCategory[];
  photos: Photo[];
}

export interface Comment {
  id: string;
  user: User;
  comment: string;
  commentPhotos: Photo[];
  productId: string;
  title: string;
  stars: number;
}

export interface Photo {
  id: string;
  photoUrl: string;
}

export interface User {
  roleId: string;
  username: string;
  passwordHash: string;
  passwordSalt: string;
  refreshToken: string;
  cartId: string;
  tokenCreated: Date;
  tokenExpires: Date;
  id: string;
}

export interface ProductCategory {
  id: string;
  name: string;
  description: string;
}
